Undo/COUndoTrack.m

```diff
@@ -566,13 +573,30 @@ - (void) doCommand: (COCommandGroup *)aCommand inverse: (BOOL)inverse addToStore
 {
 	COCommandGroup *commandToApply = (inverse ? [aCommand inverse] : aCommand);
 	[commandToApply setParentUndoTrack: self];
-	
+
 	NSMutableDictionary *md = [NSMutableDictionary new];
-	md[kCOCommitMetadataIdentifier] = inverse ? @"org.etoile.CoreObject.undo" : @"org.etoile.CoreObject.redo";
-	if ([aCommand localizedShortDescription] != nil)
+	NSString *identifier = aCommand.metadata[kCOCommitMetadataIdentifier];
+	NSString *shortDescription = aCommand.metadata[kCOCommitMetadataShortDescription];
+	NSArray *args = aCommand.metadata[kCOCommitMetadataShortDescriptionArguments];
+	NSNumber *inversedValue = aCommand.metadata[kCOCommitMetadataNodeInversed];
+	BOOL inversed = inverse && (inversedValue == nil || !inversedValue.boolValue);
```

Comment:
I'm not sure if checking aCommand.metadata[kCOCommitMetadataNodeInversed] is a good idea here.

e.g. if you have 3 commits: "A", "undo(A)", "undo(undo(A))", what you have implemented currently will give "undo(undo(A))" an kCOCommitMetadataNodeInversed value of NO, but the kCOCommitMetadataNodeUUID will be set to "undo(A)", which are contradictory (the last commit is not applying "undo(A)", but un-applying it). I think it would be better to just have `BOOL inversed = inverse;`. If the API user wants to, they can follow the chain of kCOCommitMetadataNodeUUID to find out that this selective undo is in fact re-applying an original commit.